common/static/common/js/discussion/views/discussion_thread_view.js

```diff
@@ -299,7 +305,9 @@
                 var view;
                 response.set('thread', this.model);
                 view = new ThreadResponseView($.extend({
-                    model: response
+                    model: response,
+                    mode: this.mode,
+                    start_header: this.start_header + 1
```

Comment:
This has to do with how the ThreadResponseView is displayed.  It is always shown as a "child" of the Discussion thread view.  As a result its headers should be 1 greater than those in the discussion thread view.